kroxylicious/src/main/java/io/kroxylicious/proxy/filter/multitenant/MultiTenantTransformationFilter.java

```diff
@@ -0,0 +1,224 @@
+        request.topics().forEach(topic -> applyTenantPrefix(context, topic::name, topic::setName, false));
+        request.groups().forEach(requestGroup -> requestGroup.topics().forEach(topic -> applyTenantPrefix(context, topic::name, topic::setName, false)));
+        context.forwardRequest(request);
+    }
+
+    @Override
+    public void onOffsetFetchResponse(OffsetFetchResponseData response, KrpcFilterContext context) {
+        response.topics().forEach(topic -> removeTenantPrefix(context, topic::name, topic::setName, false));
+        response.groups().forEach(responseGroup -> responseGroup.topics().forEach(topic -> removeTenantPrefix(context, topic::name, topic::setName, false)));
+        context.forwardResponse(response);
+    }
+
+    @Override
+    public void onOffsetForLeaderEpochRequest(OffsetForLeaderEpochRequestData request, KrpcFilterContext context) {
+        request.topics().forEach(topic -> applyTenantPrefix(context, topic::topic, topic::setTopic, false));
+        context.forwardRequest(request);
+    }
+
+    @Override
+    public void onOffsetForLeaderEpochResponse(OffsetForLeaderEpochResponseData response, KrpcFilterContext context) {
+        response.topics().forEach(topic -> removeTenantPrefix(context, topic::topic, topic::setTopic, false));
+        context.forwardResponse(response);
+    }
+
+    @Override
+    public void onOffsetCommitRequest(OffsetCommitRequestData request, KrpcFilterContext context) {
+        request.topics().forEach(topic -> applyTenantPrefix(context, topic::name, topic::setName, false));
+        context.forwardRequest(request);
+    }
+
+    @Override
+    public void onOffsetCommitResponse(OffsetCommitResponseData response, KrpcFilterContext context) {
+        response.topics().forEach(topic -> removeTenantPrefix(context, topic::name, topic::setName, false));
+        context.forwardResponse(response);
+    }
+
+    @Override
+    public void onFetchRequest(FetchRequestData request, KrpcFilterContext context) {
+        request.topics().forEach(topic -> applyTenantPrefix(context, topic::topic, topic::setTopic, topic.topicId() != null));
+        context.forwardRequest(request);
+    }
+
+    @Override
+    public void onFetchResponse(FetchResponseData response, KrpcFilterContext context) {
+        response.responses().forEach(topic -> removeTenantPrefix(context, topic::topic, topic::setTopic, topic.topicId() != null));
+        context.forwardResponse(response);
+
+    }
+
+    private void applyTenantPrefix(KrpcFilterContext context, Supplier<String> getter, Consumer<String> setter, boolean ignoreEmpty) {
+        var tenantPrefix = getTenantPrefix(context);
+        String clientSideName = getter.get();
+        if (ignoreEmpty && (clientSideName == null || clientSideName.isEmpty())) {
+            return;
+        }
+        setter.accept(String.format("%s-%s", tenantPrefix, clientSideName));
```

Comment:
I'd expect `String.format` will be more expensive than plain old concatention using `+`